### PR TITLE
Fix stream tests

### PR DESCRIPTION
**Pull request by lynnagara**
- Adds ability to mock jqxhr response headers
- Mock api client will always return the last added response, so single responses can be overridden without clearing all
- Update the stream tests to use the mock api client

### CODE REVIEW

File: src/sentry/static/sentry/app/__mocks__/api.jsx [hunk 1]
```diff
@@ -116,7 +117,9 @@ class Client {
           options.success,
           response.body,
           {},
-          {getResponseHeader: () => {}}
+          {
+            getResponseHeader: key => response.headers[key],
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/__mocks__/api.jsx]: 👌

File: tests/js/spec/views/accountNotifications.spec.jsx [hunk 2]
```diff
@@ -20,15 +19,10 @@ describe('AccountNotifications', function() {
         subscribeByDefault: true,
       },
     });
+  });
 
-    MockApiClient.addMockResponse({
```
Comment (lynnagara) [inline tests/js/spec/views/accountNotifications.spec.jsx]: @billyvg Do you know why was this here? It's got the same URL as the first one so it would be ignored

File: tests/js/spec/views/stream.spec.jsx [hunk 3]
```diff
@@ -19,23 +19,33 @@ const DEFAULT_LINKS_HEADER =
...
-    stubbedApiRequest = sandbox.stub(Client.prototype, 'request', (url, options) => {
-      if (
-        url === 'http://127.0.0.1/api/0/projects/sentry/ludic-science/searches/' &&
-        options.method === 'GET'
-      ) {
-        options.success &&
-          options.success([{id: '789', query: 'is:unresolved', name: 'test'}]);
-      }
-      options.complete && options.complete();
+    MockApiClient.clearMockResponses();
```
Comment (MaxBittker) [inline tests/js/spec/views/stream.spec.jsx]: tiny nit, but, afterEach the clear?

File: tests/js/spec/views/stream.spec.jsx [hunk 4]
```diff
@@ -19,23 +19,33 @@ const DEFAULT_LINKS_HEADER =
...
+    });
+    MockApiClient.addMockResponse({
+      url: '/projects/123/456/tags/',
+      body: TestStubs.Tags(),
     });
+    MockApiClient.addMockResponse({
+      url: '/projects/123/456/processingissues/',
+      method: 'GET',
+    });
+    sandbox.stub(browserHistory, 'push');
```
Comment (MaxBittker) [inline tests/js/spec/views/stream.spec.jsx]: if any of these are only used in a single test, it _may_ read better to set them there, but if not nbd

File: tests/js/spec/views/accountNotifications.spec.jsx (hunk 2 above)
Comment (billyvg) [reply tests/js/spec/views/accountNotifications.spec.jsx]: Probably just copy pasta leftovers

**Review by MaxBittker — APPROVED** [review]
✌️ 
looks like a good cleanup

File: tests/js/spec/views/stream.spec.jsx (hunk 4 above)
Comment (lynnagara) [reply tests/js/spec/views/stream.spec.jsx]: I think it's used more broadly, this component changes url to match the query frequently

File: tests/js/spec/views/stream.spec.jsx (hunk 3 above)
Comment (lynnagara) [reply tests/js/spec/views/stream.spec.jsx]: will do
